modules/platforms/dotnet/Apache.Ignite.Core/Impl/Cache/Query/FieldsQueryCursor.cs

```diff
@@ -85,5 +92,26 @@ public IList<string> FieldNames
                            Target.OutStream(OpGetFieldNames, reader => reader.ReadStringCollection())));
             }
         }
+
+        /** <inheritdoc /> */
+        public IList<IQueryCursorField> Fields 
+        {
+            get
+            {
+                if (_fieldsMeta == null)
+                {
+                    var metadata = Target.OutStream(
+                        OpGetFieldsMeta,
+                        reader => reader.ReadCollectionRaw(stream =>
+                            new QueryCursorField(stream) as IQueryCursorField))
+                        ?? new List<IQueryCursorField>();
+
+                    _fieldsMeta = new ReadOnlyCollection<IQueryCursorField>(
+                        metadata as List<IQueryCursorField> ?? metadata.ToList());
```

Comment:
`ToList` causes unnecessary allocations here. My proposal:

 * Remove `??` above, get the result of `ReadCollectionRaw` as is
* `_fieldsMeta = meta != null ? meta.AsReadOnly() : new ReadOnlyCollection()`

Reply:
As far as i understand .AsReadOnly() is not an option here cause we have IList in contract and if the underlying collection is mutable it will allow mutations.

Reply:
Changed the underlying method to return IList since it is already returning one